molo/core/api/importers.py

```
@@ -0,0 +1,231 @@
+        """
+        Sections can have SectionPage and ArticlePage child objects.
+        These have different fields, and thus have to be treated
+        differently.
+        """
+        # assemble url
+        base_url = base_url.rstrip("/")
+        url = base_url + API_PAGES_ENDPOINT + "?type=" + self._content_type + \
+            "&fields=" + ",".join(self._fields) + \
+            "&order=latest_revision_created_at"
+
+        # make request
+        try:
+            response = requests.get(url)
+            self._base_url = base_url
+            self._content = response.json()
+            self._content = self._content["items"]
+            return self._content
+        except requests.exceptions.ConnectionError:
+            return "No content could be found from {}. " \
+                "Are you sure this is the correct URL?".format(base_url)
+        except requests.exceptions.RequestException:
+            return "Content could not be imported at this time. " \
+                   "Please try again later."
+
+    def content(self):
+        if self._content:
+            return self._content
+        return []
+
+    def save(self, indexes, parent_id):
+        pass
+
+
+class ArticlePageImporter(PageImporter):
+
+    def __init__(self, base_url=None, content=None, content_type=None):
+        super(ArticlePageImporter, self).__init__(
+            base_url=base_url, content=content, content_type=content_type
+        )
+        self._content_type = "core.ArticlePage"
+        self._fields = ArticlePage.get_api_fields()
+
+    def save(self, indexes, parent_id):
+        if self.content():
+            parent = Page.objects.get(id=parent_id)
+            for index in indexes:
+                # Remove "id" and "meta" fields
+                selected_article = self.content()[index]
+                fields, nested_fields = separate_fields(selected_article)
+                article = ArticlePage(**fields)
+
+                # TODO: u'related_sections'
```

We can make a consideration that `related_sections` can fail silently if they are not being imported.